Crash after creating a new site

Crash after creating a new site (it's probably unrelated) and going back to "My Site"
```
  Process: org.wordpress.android.beta, PID: 10377
  org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
      at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:505)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:487)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479)
      at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67)
      at android.os.Handler.dispatchMessage(Handler.java:102)
      at android.os.Looper.loop(Looper.java:154)
      at android.app.ActivityThread.main(ActivityThread.java:6077)
      at java.lang.reflect.Method.invoke(Native Method)
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
  Caused by: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@b89bfef rejected from java.util.concurrent.ScheduledThreadPoolExecutor@3fbbafc[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]                                            
      at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2049)
      at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:814)
      at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:305)
      at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
      at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:652)
      at org.wordpress.android.util.helpers.Debouncer.debounce(Debouncer.java:18)
      at org.wordpress.android.ui.main.SitePickerActivity.onSiteChanged(SitePickerActivity.java:207)
      at java.lang.reflect.Method.invoke(Native Method)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
      at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:479) 
      at org.greenrobot.eventbus.HandlerPoster.handleMessage(HandlerPoster.java:67) 
      at android.os.Handler.dispatchMessage(Handler.java:102) 
      at android.os.Looper.loop(Looper.java:154) 
      at android.app.ActivityThread.main(ActivityThread.java:6077) 
      at java.lang.reflect.Method.invoke(Native Method) 
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865) 
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755) 
```

--- a/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/helpers/Debouncer.java
+++ b/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/helpers/Debouncer.java
@@ -15,6 +15,9 @@ public class Debouncer {
      * or cancels its execution if the method is called with the same key within the {@code delay} again.
      */
     public void debounce(final Object key, final Runnable runnable, long delay, TimeUnit unit) {
+        if (mScheduler.isShutdown()) {
+            return;
+        }
         final Future<?> prev = mDelayedMap.put(key, mScheduler.schedule(new Runnable() {
             @Override
             public void run() {

